{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "60dcd7e2",
   "metadata": {},
   "source": [
    "### conda env\n",
    "\n",
    "```conda create -n gih jupyter pandas numpy matplotlib seaborn trim-galore```\n",
    "\n",
    "```conda activate gih```"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "703b8d88",
   "metadata": {},
   "source": [
    "## Download read files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7c1c7246",
   "metadata": {},
   "outputs": [],
   "source": [
    "!fastq-dump --gzip SRR6908978 SRR6908979 SRR6908980 SRR6908981 SRR6908982 SRR6908983 SRR6908984 SRR6908985 SRR6908986 SRR6908987 SRR6908988 SRR6908989"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "12222f5d",
   "metadata": {},
   "source": [
    "## Reference = Fusarium graminearum Tri6 gene\n",
    "\n",
    "- https://www.ncbi.nlm.nih.gov/nuccore/AB017495.1"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c185420d",
   "metadata": {},
   "source": [
    "### Notes:\n",
    "- can align to a whole transcriptome or genome, but Tri6 is the target of the dsRNA, so makes sense to use that in this instance.\n",
    "- normalisation if used is based on library size\n",
    "- pre-filtering library is experiment specific - i.e. a mixed fungal / plant sample - will leave unfiltered in this case.\n",
    "- packages such as BBDuk are good for filtering if needed"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "118c9305",
   "metadata": {},
   "source": [
    "## Trim adapters"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "1a0ea0e5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Using an excessive number of cores has a diminishing return! It is recommended not to exceed 8 cores per trimming process (you asked for 40 cores). Please consider re-specifying\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 4.1\n",
      "Could not detect version of Python used by Cutadapt from the first line of Cutadapt (but found this: >>>#!/bin/sh<<<)\n",
      "Letting the (modified) Cutadapt deal with the Python version instead\n",
      "pigz 2.6\n",
      "Parallel gzip (pigz) detected. Proceeding with multicore (de)compression using 40 cores\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Output will be written into the directory: /home/steve/Dropbox/bioinf/gih/reads/processed/\n",
      "Reducing length cutoff to 18bp for small RNA-Seq reads because a cutoff of 20bp may remove some short species of small RNAs if they had been trimmed by 1,2 or 3bp\n",
      "Writing report to '/home/steve/Dropbox/bioinf/gih/reads/processed/SRR6908985.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: reads/raw/SRR6908985.fastq.gz\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.6.7\n",
      "Cutadapt version: 4.1\n",
      "Python version: could not detect\n",
      "Number of cores used for trimming: 40\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'TGGAATTCTCGG' (Illumina small RNA adapter; user defined)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 18 bp\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Cutadapt seems to be fairly up-to-date (version 4.1). Setting -j 40\n",
      "Writing final adapter and quality trimmed output to SRR6908985_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'TGGAATTCTCGG' from file reads/raw/SRR6908985.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "This is cutadapt 4.1 with Python 3.9.13\n",
      "Command line parameters: -j 40 -e 0.1 -q 20 -O 1 -a TGGAATTCTCGG reads/raw/SRR6908985.fastq.gz\n",
      "Processing single-end reads on 40 cores ...\n",
      "Finished in 29.59 s (2 µs/read; 38.05 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              18,764,769\n",
      "Reads with adapters:                18,614,275 (99.2%)\n",
      "Reads written (passing filters):    18,764,769 (100.0%)\n",
      "\n",
      "Total basepairs processed:   957,003,219 bp\n",
      "Quality-trimmed:               1,197,270 bp (0.1%)\n",
      "Total written (filtered):    504,996,476 bp (52.8%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: TGGAATTCTCGG; Type: regular 3'; Length: 12; Trimmed: 18614275 times\n",
      "\n",
      "Minimum overlap: 1\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-12 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 16.7%\n",
      "  C: 44.2%\n",
      "  G: 13.4%\n",
      "  T: 25.5%\n",
      "  none/other: 0.2%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t83365\t4691192.2\t0\t83365\n",
      "2\t41349\t1172798.1\t0\t41349\n",
      "3\t47804\t293199.5\t0\t47804\n",
      "4\t46202\t73299.9\t0\t46202\n",
      "5\t59071\t18325.0\t0\t59071\n",
      "6\t74147\t4581.2\t0\t74147\n",
      "7\t102435\t1145.3\t0\t102435\n",
      "8\t169514\t286.3\t0\t169514\n",
      "9\t195995\t71.6\t0\t195152 843\n",
      "10\t292185\t17.9\t1\t285852 6333\n",
      "11\t419722\t4.5\t1\t412032 7690\n",
      "12\t282979\t1.1\t1\t277203 5776\n",
      "13\t274413\t1.1\t1\t268349 6064\n",
      "14\t265811\t1.1\t1\t260628 5183\n",
      "15\t382958\t1.1\t1\t375342 7616\n",
      "16\t367123\t1.1\t1\t359267 7856\n",
      "17\t1090732\t1.1\t1\t1070431 20301\n",
      "18\t511262\t1.1\t1\t500063 11199\n",
      "19\t596961\t1.1\t1\t583804 13157\n",
      "20\t645010\t1.1\t1\t630845 14165\n",
      "21\t1069809\t1.1\t1\t1049325 20484\n",
      "22\t1053958\t1.1\t1\t1033231 20727\n",
      "23\t1088020\t1.1\t1\t1065206 22814\n",
      "24\t927713\t1.1\t1\t909597 18116\n",
      "25\t744428\t1.1\t1\t729180 15248\n",
      "26\t687561\t1.1\t1\t671791 15770\n",
      "27\t635712\t1.1\t1\t619157 16555\n",
      "28\t569414\t1.1\t1\t555656 13758\n",
      "29\t687006\t1.1\t1\t672805 14201\n",
      "30\t664398\t1.1\t1\t651089 13309\n",
      "31\t484982\t1.1\t1\t475562 9420\n",
      "32\t690023\t1.1\t1\t676662 13361\n",
      "33\t439354\t1.1\t1\t430265 9089\n",
      "34\t518176\t1.1\t1\t507380 10796\n",
      "35\t380737\t1.1\t1\t372969 7768\n",
      "36\t304150\t1.1\t1\t297733 6417\n",
      "37\t316474\t1.1\t1\t310040 6434\n",
      "38\t323393\t1.1\t1\t316975 6418\n",
      "39\t407903\t1.1\t1\t400443 7460\n",
      "40\t242961\t1.1\t1\t238366 4595\n",
      "41\t163228\t1.1\t1\t160412 2816\n",
      "42\t93443\t1.1\t1\t91596 1847\n",
      "43\t86498\t1.1\t1\t84886 1612\n",
      "44\t35634\t1.1\t1\t34922 712\n",
      "45\t10161\t1.1\t1\t9906 255\n",
      "46\t2736\t1.1\t1\t2675 61\n",
      "47\t754\t1.1\t1\t731 23\n",
      "48\t584\t1.1\t1\t563 21\n",
      "49\t1339\t1.1\t1\t1296 43\n",
      "50\t1934\t1.1\t1\t1854 80\n",
      "51\t32754\t1.1\t1\t31819 935\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: reads/raw/SRR6908985.fastq.gz\n",
      "=============================================\n",
      "18764769 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 18 bp:\t2939902 (15.7%)\n",
      "\n",
      "Using an excessive number of cores has a diminishing return! It is recommended not to exceed 8 cores per trimming process (you asked for 40 cores). Please consider re-specifying\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 4.1\n",
      "Could not detect version of Python used by Cutadapt from the first line of Cutadapt (but found this: >>>#!/bin/sh<<<)\n",
      "Letting the (modified) Cutadapt deal with the Python version instead\n",
      "pigz 2.6\n",
      "Parallel gzip (pigz) detected. Proceeding with multicore (de)compression using 40 cores\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Output will be written into the directory: /home/steve/Dropbox/bioinf/gih/reads/processed/\n",
      "Reducing length cutoff to 18bp for small RNA-Seq reads because a cutoff of 20bp may remove some short species of small RNAs if they had been trimmed by 1,2 or 3bp\n",
      "Writing report to '/home/steve/Dropbox/bioinf/gih/reads/processed/SRR6908987.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: reads/raw/SRR6908987.fastq.gz\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.6.7\n",
      "Cutadapt version: 4.1\n",
      "Python version: could not detect\n",
      "Number of cores used for trimming: 40\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'TGGAATTCTCGG' (Illumina small RNA adapter; user defined)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 18 bp\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Cutadapt seems to be fairly up-to-date (version 4.1). Setting -j 40\n",
      "Writing final adapter and quality trimmed output to SRR6908987_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'TGGAATTCTCGG' from file reads/raw/SRR6908987.fastq.gz <<< \n",
      "This is cutadapt 4.1 with Python 3.9.13\n",
      "Command line parameters: -j 40 -e 0.1 -q 20 -O 1 -a TGGAATTCTCGG reads/raw/SRR6908987.fastq.gz\n",
      "Processing single-end reads on 40 cores ...\n",
      "Finished in 15.72 s (2 µs/read; 35.46 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:               9,291,287\n",
      "Reads with adapters:                 9,214,433 (99.2%)\n",
      "Reads written (passing filters):     9,291,287 (100.0%)\n",
      "\n",
      "Total basepairs processed:   473,855,637 bp\n",
      "Quality-trimmed:                 329,834 bp (0.1%)\n",
      "Total written (filtered):    280,814,652 bp (59.3%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: TGGAATTCTCGG; Type: regular 3'; Length: 12; Trimmed: 9214433 times\n",
      "\n",
      "Minimum overlap: 1\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-12 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 27.3%\n",
      "  C: 33.9%\n",
      "  G: 16.2%\n",
      "  T: 22.2%\n",
      "  none/other: 0.5%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t40895\t2322821.8\t0\t40895\n",
      "2\t24750\t580705.4\t0\t24750\n",
      "3\t29142\t145176.4\t0\t29142\n",
      "4\t31360\t36294.1\t0\t31360\n",
      "5\t46682\t9073.5\t0\t46682\n",
      "6\t55335\t2268.4\t0\t55335\n",
      "7\t71197\t567.1\t0\t71197\n",
      "8\t93715\t141.8\t0\t93715\n",
      "9\t131072\t35.4\t0\t130594 478\n",
      "10\t321215\t8.9\t1\t318032 3183\n",
      "11\t283558\t2.2\t1\t280435 3123\n",
      "12\t188176\t0.6\t1\t185219 2957\n",
      "13\t233892\t0.6\t1\t230246 3646\n",
      "14\t228657\t0.6\t1\t225734 2923\n",
      "15\t356899\t0.6\t1\t352088 4811\n",
      "16\t356778\t0.6\t1\t351218 5560\n",
      "17\t1417569\t0.6\t1\t1402135 15434\n",
      "18\t382061\t0.6\t1\t377177 4884\n",
      "19\t350651\t0.6\t1\t343217 7434\n",
      "20\t328366\t0.6\t1\t321563 6803\n",
      "21\t350389\t0.6\t1\t343148 7241\n",
      "22\t336633\t0.6\t1\t329822 6811\n",
      "23\t345098\t0.6\t1\t340941 4157\n",
      "24\t313969\t0.6\t1\t310094 3875\n",
      "25\t280409\t0.6\t1\t276902 3507\n",
      "26\t283545\t0.6\t1\t280044 3501\n",
      "27\t300599\t0.6\t1\t296671 3928\n",
      "28\t278733\t0.6\t1\t272756 5977\n",
      "29\t283243\t0.6\t1\t277258 5985\n",
      "30\t251923\t0.6\t1\t248958 2965\n",
      "31\t202104\t0.6\t1\t199557 2547\n",
      "32\t191198\t0.6\t1\t188513 2685\n",
      "33\t162596\t0.6\t1\t160464 2132\n",
      "34\t134389\t0.6\t1\t132498 1891\n",
      "35\t101250\t0.6\t1\t99855 1395\n",
      "36\t84979\t0.6\t1\t83808 1171\n",
      "37\t70300\t0.6\t1\t69166 1134\n",
      "38\t51670\t0.6\t1\t50716 954\n",
      "39\t50711\t0.6\t1\t50013 698\n",
      "40\t36503\t0.6\t1\t35939 564\n",
      "41\t18551\t0.6\t1\t18251 300\n",
      "42\t17155\t0.6\t1\t16839 316\n",
      "43\t21284\t0.6\t1\t20945 339\n",
      "44\t11804\t0.6\t1\t11608 196\n",
      "45\t4598\t0.6\t1\t4512 86\n",
      "46\t4033\t0.6\t1\t3949 84\n",
      "47\t2539\t0.6\t1\t2482 57\n",
      "48\t1880\t0.6\t1\t1840 40\n",
      "49\t1788\t0.6\t1\t1745 43\n",
      "50\t2547\t0.6\t1\t2478 69\n",
      "51\t46043\t0.6\t1\t45320 723\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: reads/raw/SRR6908987.fastq.gz\n",
      "=============================================\n",
      "9291287 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 18 bp:\t665573 (7.2%)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Using an excessive number of cores has a diminishing return! It is recommended not to exceed 8 cores per trimming process (you asked for 40 cores). Please consider re-specifying\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 4.1\n",
      "Could not detect version of Python used by Cutadapt from the first line of Cutadapt (but found this: >>>#!/bin/sh<<<)\n",
      "Letting the (modified) Cutadapt deal with the Python version instead\n",
      "pigz 2.6\n",
      "Parallel gzip (pigz) detected. Proceeding with multicore (de)compression using 40 cores\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Output will be written into the directory: /home/steve/Dropbox/bioinf/gih/reads/processed/\n",
      "Reducing length cutoff to 18bp for small RNA-Seq reads because a cutoff of 20bp may remove some short species of small RNAs if they had been trimmed by 1,2 or 3bp\n",
      "Writing report to '/home/steve/Dropbox/bioinf/gih/reads/processed/SRR6908980.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: reads/raw/SRR6908980.fastq.gz\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.6.7\n",
      "Cutadapt version: 4.1\n",
      "Python version: could not detect\n",
      "Number of cores used for trimming: 40\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'TGGAATTCTCGG' (Illumina small RNA adapter; user defined)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 18 bp\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Cutadapt seems to be fairly up-to-date (version 4.1). Setting -j 40\n",
      "Writing final adapter and quality trimmed output to SRR6908980_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'TGGAATTCTCGG' from file reads/raw/SRR6908980.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "This is cutadapt 4.1 with Python 3.9.13\n",
      "Command line parameters: -j 40 -e 0.1 -q 20 -O 1 -a TGGAATTCTCGG reads/raw/SRR6908980.fastq.gz\n",
      "Processing single-end reads on 40 cores ...\n",
      "Finished in 16.65 s (2 µs/read; 36.69 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              10,180,880\n",
      "Reads with adapters:                10,095,743 (99.2%)\n",
      "Reads written (passing filters):    10,180,880 (100.0%)\n",
      "\n",
      "Total basepairs processed:   519,224,880 bp\n",
      "Quality-trimmed:                 665,020 bp (0.1%)\n",
      "Total written (filtered):    274,773,471 bp (52.9%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: TGGAATTCTCGG; Type: regular 3'; Length: 12; Trimmed: 10095743 times\n",
      "\n",
      "Minimum overlap: 1\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-12 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 16.9%\n",
      "  C: 44.1%\n",
      "  G: 13.2%\n",
      "  T: 25.5%\n",
      "  none/other: 0.3%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t48138\t2545220.0\t0\t48138\n",
      "2\t24674\t636305.0\t0\t24674\n",
      "3\t27938\t159076.2\t0\t27938\n",
      "4\t29796\t39769.1\t0\t29796\n",
      "5\t38889\t9942.3\t0\t38889\n",
      "6\t47822\t2485.6\t0\t47822\n",
      "7\t64341\t621.4\t0\t64341\n",
      "8\t100394\t155.3\t0\t100394\n",
      "9\t117781\t38.8\t0\t117312 469\n",
      "10\t164218\t9.7\t1\t160886 3332\n",
      "11\t222813\t2.4\t1\t218926 3887\n",
      "12\t155768\t0.6\t1\t152668 3100\n",
      "13\t154831\t0.6\t1\t151642 3189\n",
      "14\t148953\t0.6\t1\t146083 2870\n",
      "15\t207720\t0.6\t1\t203637 4083\n",
      "16\t204219\t0.6\t1\t200105 4114\n",
      "17\t590945\t0.6\t1\t580144 10801\n",
      "18\t279390\t0.6\t1\t273318 6072\n",
      "19\t310785\t0.6\t1\t303762 7023\n",
      "20\t358270\t0.6\t1\t351009 7261\n",
      "21\t538393\t0.6\t1\t528635 9758\n",
      "22\t518827\t0.6\t1\t508923 9904\n",
      "23\t559048\t0.6\t1\t547631 11417\n",
      "24\t480769\t0.6\t1\t471156 9613\n",
      "25\t415792\t0.6\t1\t407847 7945\n",
      "26\t409571\t0.6\t1\t400374 9197\n",
      "27\t351501\t0.6\t1\t342594 8907\n",
      "28\t328831\t0.6\t1\t321188 7643\n",
      "29\t361093\t0.6\t1\t353288 7805\n",
      "30\t363669\t0.6\t1\t356026 7643\n",
      "31\t283455\t0.6\t1\t277524 5931\n",
      "32\t359648\t0.6\t1\t352911 6737\n",
      "33\t255081\t0.6\t1\t249777 5304\n",
      "34\t285143\t0.6\t1\t279483 5660\n",
      "35\t187002\t0.6\t1\t182996 4006\n",
      "36\t187772\t0.6\t1\t183788 3984\n",
      "37\t172368\t0.6\t1\t168727 3641\n",
      "38\t157186\t0.6\t1\t154047 3139\n",
      "39\t249836\t0.6\t1\t245527 4309\n",
      "40\t121584\t0.6\t1\t119355 2229\n",
      "41\t79550\t0.6\t1\t78314 1236\n",
      "42\t36943\t0.6\t1\t36153 790\n",
      "43\t33888\t0.6\t1\t33247 641\n",
      "44\t18129\t0.6\t1\t17771 358\n",
      "45\t7907\t0.6\t1\t7739 168\n",
      "46\t2650\t0.6\t1\t2594 56\n",
      "47\t628\t0.6\t1\t612 16\n",
      "48\t527\t0.6\t1\t501 26\n",
      "49\t1379\t0.6\t1\t1326 53\n",
      "50\t1731\t0.6\t1\t1646 85\n",
      "51\t28157\t0.6\t1\t27195 962\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: reads/raw/SRR6908980.fastq.gz\n",
      "=============================================\n",
      "10180880 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 18 bp:\t1585039 (15.6%)\n",
      "\n",
      "Using an excessive number of cores has a diminishing return! It is recommended not to exceed 8 cores per trimming process (you asked for 40 cores). Please consider re-specifying\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 4.1\n",
      "Could not detect version of Python used by Cutadapt from the first line of Cutadapt (but found this: >>>#!/bin/sh<<<)\n",
      "Letting the (modified) Cutadapt deal with the Python version instead\n",
      "pigz 2.6\n",
      "Parallel gzip (pigz) detected. Proceeding with multicore (de)compression using 40 cores\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Output will be written into the directory: /home/steve/Dropbox/bioinf/gih/reads/processed/\n",
      "Reducing length cutoff to 18bp for small RNA-Seq reads because a cutoff of 20bp may remove some short species of small RNAs if they had been trimmed by 1,2 or 3bp\n",
      "Writing report to '/home/steve/Dropbox/bioinf/gih/reads/processed/SRR6908988.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: reads/raw/SRR6908988.fastq.gz\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.6.7\n",
      "Cutadapt version: 4.1\n",
      "Python version: could not detect\n",
      "Number of cores used for trimming: 40\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'TGGAATTCTCGG' (Illumina small RNA adapter; user defined)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 18 bp\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Cutadapt seems to be fairly up-to-date (version 4.1). Setting -j 40\n",
      "Writing final adapter and quality trimmed output to SRR6908988_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'TGGAATTCTCGG' from file reads/raw/SRR6908988.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "This is cutadapt 4.1 with Python 3.9.13\n",
      "Command line parameters: -j 40 -e 0.1 -q 20 -O 1 -a TGGAATTCTCGG reads/raw/SRR6908988.fastq.gz\n",
      "Processing single-end reads on 40 cores ...\n",
      "Finished in 22.08 s (2 µs/read; 37.22 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              13,698,028\n",
      "Reads with adapters:                13,586,580 (99.2%)\n",
      "Reads written (passing filters):    13,698,028 (100.0%)\n",
      "\n",
      "Total basepairs processed:   698,599,428 bp\n",
      "Quality-trimmed:                 444,760 bp (0.1%)\n",
      "Total written (filtered):    412,721,204 bp (59.1%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: TGGAATTCTCGG; Type: regular 3'; Length: 12; Trimmed: 13586580 times\n",
      "\n",
      "Minimum overlap: 1\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-12 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 27.7%\n",
      "  C: 33.4%\n",
      "  G: 14.7%\n",
      "  T: 23.8%\n",
      "  none/other: 0.5%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t56983\t3424507.0\t0\t56983\n",
      "2\t33267\t856126.8\t0\t33267\n",
      "3\t36931\t214031.7\t0\t36931\n",
      "4\t41908\t53507.9\t0\t41908\n",
      "5\t63800\t13377.0\t0\t63800\n",
      "6\t74933\t3344.2\t0\t74933\n",
      "7\t97435\t836.1\t0\t97435\n",
      "8\t135008\t209.0\t0\t135008\n",
      "9\t192481\t52.3\t0\t191872 609\n",
      "10\t453922\t13.1\t1\t449837 4085\n",
      "11\t381109\t3.3\t1\t377380 3729\n",
      "12\t268082\t0.8\t1\t264633 3449\n",
      "13\t326396\t0.8\t1\t322146 4250\n",
      "14\t332448\t0.8\t1\t328566 3882\n",
      "15\t511300\t0.8\t1\t505033 6267\n",
      "16\t493066\t0.8\t1\t485867 7199\n",
      "17\t2118438\t0.8\t1\t2098409 20029\n",
      "18\t570177\t0.8\t1\t563797 6380\n",
      "19\t534688\t0.8\t1\t524139 10549\n",
      "20\t496781\t0.8\t1\t487030 9751\n",
      "21\t513767\t0.8\t1\t503747 10020\n",
      "22\t527141\t0.8\t1\t517146 9995\n",
      "23\t562826\t0.8\t1\t556679 6147\n",
      "24\t509347\t0.8\t1\t503586 5761\n",
      "25\t433947\t0.8\t1\t429171 4776\n",
      "26\t436783\t0.8\t1\t432041 4742\n",
      "27\t458949\t0.8\t1\t453529 5420\n",
      "28\t389606\t0.8\t1\t381950 7656\n",
      "29\t401691\t0.8\t1\t393581 8110\n",
      "30\t353268\t0.8\t1\t349375 3893\n",
      "31\t287144\t0.8\t1\t283819 3325\n",
      "32\t287413\t0.8\t1\t283894 3519\n",
      "33\t237030\t0.8\t1\t234211 2819\n",
      "34\t206015\t0.8\t1\t203619 2396\n",
      "35\t141879\t0.8\t1\t140129 1750\n",
      "36\t122669\t0.8\t1\t121200 1469\n",
      "37\t105895\t0.8\t1\t104520 1375\n",
      "38\t81877\t0.8\t1\t80697 1180\n",
      "39\t83643\t0.8\t1\t82679 964\n",
      "40\t52886\t0.8\t1\t52160 726\n",
      "41\t24927\t0.8\t1\t24571 356\n",
      "42\t28915\t0.8\t1\t28530 385\n",
      "43\t26739\t0.8\t1\t26383 356\n",
      "44\t13352\t0.8\t1\t13157 195\n",
      "45\t4805\t0.8\t1\t4740 65\n",
      "46\t3858\t0.8\t1\t3811 47\n",
      "47\t2310\t0.8\t1\t2259 51\n",
      "48\t1563\t0.8\t1\t1517 46\n",
      "49\t1981\t0.8\t1\t1937 44\n",
      "50\t2751\t0.8\t1\t2654 97\n",
      "51\t62450\t0.8\t1\t61512 938\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: reads/raw/SRR6908988.fastq.gz\n",
      "=============================================\n",
      "13698028 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 18 bp:\t976151 (7.1%)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Using an excessive number of cores has a diminishing return! It is recommended not to exceed 8 cores per trimming process (you asked for 40 cores). Please consider re-specifying\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 4.1\n",
      "Could not detect version of Python used by Cutadapt from the first line of Cutadapt (but found this: >>>#!/bin/sh<<<)\n",
      "Letting the (modified) Cutadapt deal with the Python version instead\n",
      "pigz 2.6\n",
      "Parallel gzip (pigz) detected. Proceeding with multicore (de)compression using 40 cores\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Output will be written into the directory: /home/steve/Dropbox/bioinf/gih/reads/processed/\n",
      "Reducing length cutoff to 18bp for small RNA-Seq reads because a cutoff of 20bp may remove some short species of small RNAs if they had been trimmed by 1,2 or 3bp\n",
      "Writing report to '/home/steve/Dropbox/bioinf/gih/reads/processed/SRR6908986.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: reads/raw/SRR6908986.fastq.gz\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.6.7\n",
      "Cutadapt version: 4.1\n",
      "Python version: could not detect\n",
      "Number of cores used for trimming: 40\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'TGGAATTCTCGG' (Illumina small RNA adapter; user defined)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 18 bp\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Cutadapt seems to be fairly up-to-date (version 4.1). Setting -j 40\n",
      "Writing final adapter and quality trimmed output to SRR6908986_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'TGGAATTCTCGG' from file reads/raw/SRR6908986.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "This is cutadapt 4.1 with Python 3.9.13\n",
      "Command line parameters: -j 40 -e 0.1 -q 20 -O 1 -a TGGAATTCTCGG reads/raw/SRR6908986.fastq.gz\n",
      "Processing single-end reads on 40 cores ...\n",
      "Finished in 21.93 s (2 µs/read; 37.40 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              13,667,369\n",
      "Reads with adapters:                13,575,941 (99.3%)\n",
      "Reads written (passing filters):    13,667,369 (100.0%)\n",
      "\n",
      "Total basepairs processed:   697,035,819 bp\n",
      "Quality-trimmed:               1,095,475 bp (0.2%)\n",
      "Total written (filtered):    376,963,006 bp (54.1%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: TGGAATTCTCGG; Type: regular 3'; Length: 12; Trimmed: 13575941 times\n",
      "\n",
      "Minimum overlap: 1\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-12 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 24.3%\n",
      "  C: 32.1%\n",
      "  G: 15.6%\n",
      "  T: 26.0%\n",
      "  none/other: 2.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t36079\t3416842.2\t0\t36079\n",
      "2\t13129\t854210.6\t0\t13129\n",
      "3\t10782\t213552.6\t0\t10782\n",
      "4\t12087\t53388.2\t0\t12087\n",
      "5\t18366\t13347.0\t0\t18366\n",
      "6\t27702\t3336.8\t0\t27702\n",
      "7\t45812\t834.2\t0\t45812\n",
      "8\t87747\t208.5\t0\t87747\n",
      "9\t112958\t52.1\t0\t112482 476\n",
      "10\t188993\t13.0\t1\t184157 4836\n",
      "11\t210450\t3.3\t1\t206148 4302\n",
      "12\t194262\t0.8\t1\t190225 4037\n",
      "13\t238431\t0.8\t1\t233343 5088\n",
      "14\t268805\t0.8\t1\t263182 5623\n",
      "15\t390529\t0.8\t1\t383183 7346\n",
      "16\t399830\t0.8\t1\t391220 8610\n",
      "17\t992552\t0.8\t1\t974697 17855\n",
      "18\t539270\t0.8\t1\t528391 10879\n",
      "19\t540487\t0.8\t1\t529438 11049\n",
      "20\t626489\t0.8\t1\t615581 10908\n",
      "21\t683613\t0.8\t1\t672916 10697\n",
      "22\t701204\t0.8\t1\t689994 11210\n",
      "23\t827299\t0.8\t1\t812914 14385\n",
      "24\t722084\t0.8\t1\t709763 12321\n",
      "25\t616756\t0.8\t1\t606586 10170\n",
      "26\t644385\t0.8\t1\t632973 11412\n",
      "27\t616787\t0.8\t1\t604802 11985\n",
      "28\t508688\t0.8\t1\t499245 9443\n",
      "29\t586084\t0.8\t1\t577128 8956\n",
      "30\t499213\t0.8\t1\t491705 7508\n",
      "31\t365283\t0.8\t1\t359127 6156\n",
      "32\t381385\t0.8\t1\t375098 6287\n",
      "33\t267232\t0.8\t1\t262346 4886\n",
      "34\t221944\t0.8\t1\t217710 4234\n",
      "35\t158319\t0.8\t1\t155216 3103\n",
      "36\t126786\t0.8\t1\t124225 2561\n",
      "37\t118961\t0.8\t1\t116573 2388\n",
      "38\t85712\t0.8\t1\t83934 1778\n",
      "39\t86096\t0.8\t1\t84538 1558\n",
      "40\t45659\t0.8\t1\t44794 865\n",
      "41\t29916\t0.8\t1\t29435 481\n",
      "42\t20090\t0.8\t1\t19695 395\n",
      "43\t19535\t0.8\t1\t19226 309\n",
      "44\t9650\t0.8\t1\t9486 164\n",
      "45\t3438\t0.8\t1\t3373 65\n",
      "46\t1560\t0.8\t1\t1526 34\n",
      "47\t1015\t0.8\t1\t981 34\n",
      "48\t1817\t0.8\t1\t1759 58\n",
      "49\t3149\t0.8\t1\t2993 156\n",
      "50\t14136\t0.8\t1\t13651 485\n",
      "51\t253385\t0.8\t1\t248579 4806\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: reads/raw/SRR6908986.fastq.gz\n",
      "=============================================\n",
      "13667369 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 18 bp:\t1211252 (8.9%)\n",
      "\n",
      "Using an excessive number of cores has a diminishing return! It is recommended not to exceed 8 cores per trimming process (you asked for 40 cores). Please consider re-specifying\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 4.1\n",
      "Could not detect version of Python used by Cutadapt from the first line of Cutadapt (but found this: >>>#!/bin/sh<<<)\n",
      "Letting the (modified) Cutadapt deal with the Python version instead\n",
      "pigz 2.6\n",
      "Parallel gzip (pigz) detected. Proceeding with multicore (de)compression using 40 cores\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Output will be written into the directory: /home/steve/Dropbox/bioinf/gih/reads/processed/\n",
      "Reducing length cutoff to 18bp for small RNA-Seq reads because a cutoff of 20bp may remove some short species of small RNAs if they had been trimmed by 1,2 or 3bp\n",
      "Writing report to '/home/steve/Dropbox/bioinf/gih/reads/processed/SRR6908981.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: reads/raw/SRR6908981.fastq.gz\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.6.7\n",
      "Cutadapt version: 4.1\n",
      "Python version: could not detect\n",
      "Number of cores used for trimming: 40\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'TGGAATTCTCGG' (Illumina small RNA adapter; user defined)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 18 bp\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Cutadapt seems to be fairly up-to-date (version 4.1). Setting -j 40\n",
      "Writing final adapter and quality trimmed output to SRR6908981_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'TGGAATTCTCGG' from file reads/raw/SRR6908981.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "This is cutadapt 4.1 with Python 3.9.13\n",
      "Command line parameters: -j 40 -e 0.1 -q 20 -O 1 -a TGGAATTCTCGG reads/raw/SRR6908981.fastq.gz\n",
      "Processing single-end reads on 40 cores ...\n",
      "Finished in 27.72 s (2 µs/read; 39.02 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              18,023,847\n",
      "Reads with adapters:                17,885,393 (99.2%)\n",
      "Reads written (passing filters):    18,023,847 (100.0%)\n",
      "\n",
      "Total basepairs processed:   919,216,197 bp\n",
      "Quality-trimmed:               1,094,904 bp (0.1%)\n",
      "Total written (filtered):    478,603,861 bp (52.1%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: TGGAATTCTCGG; Type: regular 3'; Length: 12; Trimmed: 17885393 times\n",
      "\n",
      "Minimum overlap: 1\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-12 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 15.8%\n",
      "  C: 48.1%\n",
      "  G: 11.6%\n",
      "  T: 24.1%\n",
      "  none/other: 0.4%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t60235\t4505961.8\t0\t60235\n",
      "2\t24548\t1126490.4\t0\t24548\n",
      "3\t23744\t281622.6\t0\t23744\n",
      "4\t23309\t70405.7\t0\t23309\n",
      "5\t29618\t17601.4\t0\t29618\n",
      "6\t36719\t4400.4\t0\t36719\n",
      "7\t51074\t1100.1\t0\t51074\n",
      "8\t87017\t275.0\t0\t87017\n",
      "9\t107076\t68.8\t0\t106607 469\n",
      "10\t151604\t17.2\t1\t148025 3579\n",
      "11\t250153\t4.3\t1\t245810 4343\n",
      "12\t174678\t1.1\t1\t171315 3363\n",
      "13\t171400\t1.1\t1\t167503 3897\n",
      "14\t177864\t1.1\t1\t174458 3406\n",
      "15\t266486\t1.1\t1\t261310 5176\n",
      "16\t280112\t1.1\t1\t274569 5543\n",
      "17\t852902\t1.1\t1\t838077 14825\n",
      "18\t444626\t1.1\t1\t435022 9604\n",
      "19\t538193\t1.1\t1\t526032 12161\n",
      "20\t827020\t1.1\t1\t811018 16002\n",
      "21\t1443980\t1.1\t1\t1421332 22648\n",
      "22\t1336607\t1.1\t1\t1314480 22127\n",
      "23\t1276258\t1.1\t1\t1253225 23033\n",
      "24\t1066256\t1.1\t1\t1048873 17383\n",
      "25\t859145\t1.1\t1\t845118 14027\n",
      "26\t825226\t1.1\t1\t809705 15521\n",
      "27\t856338\t1.1\t1\t839462 16876\n",
      "28\t591580\t1.1\t1\t580061 11519\n",
      "29\t736350\t1.1\t1\t724263 12087\n",
      "30\t644096\t1.1\t1\t633235 10861\n",
      "31\t457803\t1.1\t1\t449442 8361\n",
      "32\t692771\t1.1\t1\t681748 11023\n",
      "33\t391812\t1.1\t1\t384600 7212\n",
      "34\t372428\t1.1\t1\t365548 6880\n",
      "35\t268686\t1.1\t1\t263367 5319\n",
      "36\t229537\t1.1\t1\t225077 4460\n",
      "37\t204357\t1.1\t1\t200388 3969\n",
      "38\t197756\t1.1\t1\t194034 3722\n",
      "39\t315228\t1.1\t1\t309463 5765\n",
      "40\t182004\t1.1\t1\t178820 3184\n",
      "41\t108803\t1.1\t1\t106931 1872\n",
      "42\t75122\t1.1\t1\t73831 1291\n",
      "43\t57902\t1.1\t1\t56922 980\n",
      "44\t27143\t1.1\t1\t26668 475\n",
      "45\t10301\t1.1\t1\t10123 178\n",
      "46\t3547\t1.1\t1\t3484 63\n",
      "47\t824\t1.1\t1\t800 24\n",
      "48\t875\t1.1\t1\t850 25\n",
      "49\t2176\t1.1\t1\t2111 65\n",
      "50\t3656\t1.1\t1\t3515 141\n",
      "51\t68448\t1.1\t1\t66729 1719\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: reads/raw/SRR6908981.fastq.gz\n",
      "=============================================\n",
      "18023847 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 18 bp:\t2139827 (11.9%)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Using an excessive number of cores has a diminishing return! It is recommended not to exceed 8 cores per trimming process (you asked for 40 cores). Please consider re-specifying\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 4.1\n",
      "Could not detect version of Python used by Cutadapt from the first line of Cutadapt (but found this: >>>#!/bin/sh<<<)\n",
      "Letting the (modified) Cutadapt deal with the Python version instead\n",
      "pigz 2.6\n",
      "Parallel gzip (pigz) detected. Proceeding with multicore (de)compression using 40 cores\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Output will be written into the directory: /home/steve/Dropbox/bioinf/gih/reads/processed/\n",
      "Reducing length cutoff to 18bp for small RNA-Seq reads because a cutoff of 20bp may remove some short species of small RNAs if they had been trimmed by 1,2 or 3bp\n",
      "Writing report to '/home/steve/Dropbox/bioinf/gih/reads/processed/SRR6908983.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: reads/raw/SRR6908983.fastq.gz\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.6.7\n",
      "Cutadapt version: 4.1\n",
      "Python version: could not detect\n",
      "Number of cores used for trimming: 40\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'TGGAATTCTCGG' (Illumina small RNA adapter; user defined)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 18 bp\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Cutadapt seems to be fairly up-to-date (version 4.1). Setting -j 40\n",
      "Writing final adapter and quality trimmed output to SRR6908983_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'TGGAATTCTCGG' from file reads/raw/SRR6908983.fastq.gz <<< \n",
      "This is cutadapt 4.1 with Python 3.9.13\n",
      "Command line parameters: -j 40 -e 0.1 -q 20 -O 1 -a TGGAATTCTCGG reads/raw/SRR6908983.fastq.gz\n",
      "Processing single-end reads on 40 cores ...\n",
      "Finished in 14.16 s (2 µs/read; 37.07 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:               8,749,786\n",
      "Reads with adapters:                 8,638,226 (98.7%)\n",
      "Reads written (passing filters):     8,749,786 (100.0%)\n",
      "\n",
      "Total basepairs processed:   446,239,086 bp\n",
      "Quality-trimmed:                 654,724 bp (0.1%)\n",
      "Total written (filtered):    249,142,161 bp (55.8%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: TGGAATTCTCGG; Type: regular 3'; Length: 12; Trimmed: 8638226 times\n",
      "\n",
      "Minimum overlap: 1\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-12 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 17.5%\n",
      "  C: 45.3%\n",
      "  G: 13.2%\n",
      "  T: 23.4%\n",
      "  none/other: 0.5%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t75289\t2187446.5\t0\t75289\n",
      "2\t44896\t546861.6\t0\t44896\n",
      "3\t48696\t136715.4\t0\t48696\n",
      "4\t45266\t34178.9\t0\t45266\n",
      "5\t54313\t8544.7\t0\t54313\n",
      "6\t59913\t2136.2\t0\t59913\n",
      "7\t71913\t534.0\t0\t71913\n",
      "8\t114434\t133.5\t0\t114434\n",
      "9\t141317\t33.4\t0\t140706 611\n",
      "10\t227287\t8.3\t1\t222934 4353\n",
      "11\t275511\t2.1\t1\t271144 4367\n",
      "12\t169608\t0.5\t1\t166413 3195\n",
      "13\t152591\t0.5\t1\t149388 3203\n",
      "14\t134917\t0.5\t1\t132490 2427\n",
      "15\t186388\t0.5\t1\t183104 3284\n",
      "16\t224725\t0.5\t1\t220339 4386\n",
      "17\t505035\t0.5\t1\t496150 8885\n",
      "18\t223384\t0.5\t1\t218576 4808\n",
      "19\t219420\t0.5\t1\t214530 4890\n",
      "20\t225445\t0.5\t1\t220686 4759\n",
      "21\t377602\t0.5\t1\t371129 6473\n",
      "22\t441495\t0.5\t1\t433747 7748\n",
      "23\t563775\t0.5\t1\t553234 10541\n",
      "24\t545824\t0.5\t1\t536161 9663\n",
      "25\t375916\t0.5\t1\t369339 6577\n",
      "26\t431474\t0.5\t1\t423879 7595\n",
      "27\t246658\t0.5\t1\t241007 5651\n",
      "28\t243131\t0.5\t1\t238033 5098\n",
      "29\t330600\t0.5\t1\t324897 5703\n",
      "30\t249331\t0.5\t1\t245022 4309\n",
      "31\t175120\t0.5\t1\t171795 3325\n",
      "32\t223853\t0.5\t1\t219797 4056\n",
      "33\t147015\t0.5\t1\t144152 2863\n",
      "34\t187246\t0.5\t1\t183758 3488\n",
      "35\t113475\t0.5\t1\t111221 2254\n",
      "36\t96469\t0.5\t1\t94475 1994\n",
      "37\t104358\t0.5\t1\t102253 2105\n",
      "38\t104340\t0.5\t1\t102130 2210\n",
      "39\t199204\t0.5\t1\t195573 3631\n",
      "40\t91661\t0.5\t1\t89941 1720\n",
      "41\t58899\t0.5\t1\t57979 920\n",
      "42\t27045\t0.5\t1\t26417 628\n",
      "43\t38663\t0.5\t1\t37991 672\n",
      "44\t14863\t0.5\t1\t14595 268\n",
      "45\t5109\t0.5\t1\t5010 99\n",
      "46\t1357\t0.5\t1\t1334 23\n",
      "47\t459\t0.5\t1\t446 13\n",
      "48\t424\t0.5\t1\t398 26\n",
      "49\t1071\t0.5\t1\t1022 49\n",
      "50\t2425\t0.5\t1\t2322 103\n",
      "51\t39016\t0.5\t1\t38093 923\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: reads/raw/SRR6908983.fastq.gz\n",
      "=============================================\n",
      "8749786 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 18 bp:\t1092522 (12.5%)\n",
      "\n",
      "Using an excessive number of cores has a diminishing return! It is recommended not to exceed 8 cores per trimming process (you asked for 40 cores). Please consider re-specifying\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 4.1\n",
      "Could not detect version of Python used by Cutadapt from the first line of Cutadapt (but found this: >>>#!/bin/sh<<<)\n",
      "Letting the (modified) Cutadapt deal with the Python version instead\n",
      "pigz 2.6\n",
      "Parallel gzip (pigz) detected. Proceeding with multicore (de)compression using 40 cores\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Output will be written into the directory: /home/steve/Dropbox/bioinf/gih/reads/processed/\n",
      "Reducing length cutoff to 18bp for small RNA-Seq reads because a cutoff of 20bp may remove some short species of small RNAs if they had been trimmed by 1,2 or 3bp\n",
      "Writing report to '/home/steve/Dropbox/bioinf/gih/reads/processed/SRR6908982.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: reads/raw/SRR6908982.fastq.gz\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.6.7\n",
      "Cutadapt version: 4.1\n",
      "Python version: could not detect\n",
      "Number of cores used for trimming: 40\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'TGGAATTCTCGG' (Illumina small RNA adapter; user defined)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 18 bp\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Cutadapt seems to be fairly up-to-date (version 4.1). Setting -j 40\n",
      "Writing final adapter and quality trimmed output to SRR6908982_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'TGGAATTCTCGG' from file reads/raw/SRR6908982.fastq.gz <<< \n",
      "This is cutadapt 4.1 with Python 3.9.13\n",
      "Command line parameters: -j 40 -e 0.1 -q 20 -O 1 -a TGGAATTCTCGG reads/raw/SRR6908982.fastq.gz\n",
      "Processing single-end reads on 40 cores ...\n",
      "Finished in 14.51 s (2 µs/read; 36.79 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:               8,899,438\n",
      "Reads with adapters:                 8,760,016 (98.4%)\n",
      "Reads written (passing filters):     8,899,438 (100.0%)\n",
      "\n",
      "Total basepairs processed:   453,871,338 bp\n",
      "Quality-trimmed:                 605,265 bp (0.1%)\n",
      "Total written (filtered):    271,180,056 bp (59.7%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: TGGAATTCTCGG; Type: regular 3'; Length: 12; Trimmed: 8760016 times\n",
      "\n",
      "Minimum overlap: 1\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-12 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 20.1%\n",
      "  C: 39.5%\n",
      "  G: 14.8%\n",
      "  T: 25.2%\n",
      "  none/other: 0.3%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t95383\t2224859.5\t0\t95383\n",
      "2\t63046\t556214.9\t0\t63046\n",
      "3\t70985\t139053.7\t0\t70985\n",
      "4\t68446\t34763.4\t0\t68446\n",
      "5\t86693\t8690.9\t0\t86693\n",
      "6\t94876\t2172.7\t0\t94876\n",
      "7\t109576\t543.2\t0\t109576\n",
      "8\t139503\t135.8\t0\t139503\n",
      "9\t169050\t33.9\t0\t168388 662\n",
      "10\t268106\t8.5\t1\t261546 6560\n",
      "11\t249120\t2.1\t1\t243721 5399\n",
      "12\t195261\t0.5\t1\t190624 4637\n",
      "13\t224414\t0.5\t1\t218978 5436\n",
      "14\t204313\t0.5\t1\t200131 4182\n",
      "15\t267192\t0.5\t1\t261604 5588\n",
      "16\t266979\t0.5\t1\t259820 7159\n",
      "17\t785469\t0.5\t1\t769067 16402\n",
      "18\t332579\t0.5\t1\t324564 8015\n",
      "19\t321465\t0.5\t1\t313396 8069\n",
      "20\t303808\t0.5\t1\t296526 7282\n",
      "21\t391704\t0.5\t1\t383580 8124\n",
      "22\t399322\t0.5\t1\t391392 7930\n",
      "23\t426295\t0.5\t1\t416688 9607\n",
      "24\t372896\t0.5\t1\t364757 8139\n",
      "25\t311472\t0.5\t1\t304703 6769\n",
      "26\t273097\t0.5\t1\t266677 6420\n",
      "27\t274561\t0.5\t1\t267065 7496\n",
      "28\t243521\t0.5\t1\t237380 6141\n",
      "29\t279474\t0.5\t1\t273340 6134\n",
      "30\t226115\t0.5\t1\t221434 4681\n",
      "31\t177596\t0.5\t1\t173725 3871\n",
      "32\t224010\t0.5\t1\t219326 4684\n",
      "33\t147281\t0.5\t1\t144089 3192\n",
      "34\t144776\t0.5\t1\t141236 3540\n",
      "35\t89419\t0.5\t1\t87185 2234\n",
      "36\t82293\t0.5\t1\t80299 1994\n",
      "37\t67734\t0.5\t1\t66161 1573\n",
      "38\t61948\t0.5\t1\t60522 1426\n",
      "39\t93189\t0.5\t1\t91115 2074\n",
      "40\t44180\t0.5\t1\t43183 997\n",
      "41\t25062\t0.5\t1\t24511 551\n",
      "42\t21664\t0.5\t1\t21226 438\n",
      "43\t19652\t0.5\t1\t19227 425\n",
      "44\t10505\t0.5\t1\t10285 220\n",
      "45\t3800\t0.5\t1\t3716 84\n",
      "46\t1326\t0.5\t1\t1286 40\n",
      "47\t633\t0.5\t1\t603 30\n",
      "48\t564\t0.5\t1\t539 25\n",
      "49\t1037\t0.5\t1\t995 42\n",
      "50\t1707\t0.5\t1\t1619 88\n",
      "51\t26919\t0.5\t1\t26219 700\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: reads/raw/SRR6908982.fastq.gz\n",
      "=============================================\n",
      "8899438 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 18 bp:\t701423 (7.9%)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Using an excessive number of cores has a diminishing return! It is recommended not to exceed 8 cores per trimming process (you asked for 40 cores). Please consider re-specifying\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 4.1\n",
      "Could not detect version of Python used by Cutadapt from the first line of Cutadapt (but found this: >>>#!/bin/sh<<<)\n",
      "Letting the (modified) Cutadapt deal with the Python version instead\n",
      "pigz 2.6\n",
      "Parallel gzip (pigz) detected. Proceeding with multicore (de)compression using 40 cores\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Output will be written into the directory: /home/steve/Dropbox/bioinf/gih/reads/processed/\n",
      "Reducing length cutoff to 18bp for small RNA-Seq reads because a cutoff of 20bp may remove some short species of small RNAs if they had been trimmed by 1,2 or 3bp\n",
      "Writing report to '/home/steve/Dropbox/bioinf/gih/reads/processed/SRR6908979.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: reads/raw/SRR6908979.fastq.gz\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.6.7\n",
      "Cutadapt version: 4.1\n",
      "Python version: could not detect\n",
      "Number of cores used for trimming: 40\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'TGGAATTCTCGG' (Illumina small RNA adapter; user defined)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 18 bp\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Cutadapt seems to be fairly up-to-date (version 4.1). Setting -j 40\n",
      "Writing final adapter and quality trimmed output to SRR6908979_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'TGGAATTCTCGG' from file reads/raw/SRR6908979.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "20000000 sequences processed\n",
      "This is cutadapt 4.1 with Python 3.9.13\n",
      "Command line parameters: -j 40 -e 0.1 -q 20 -O 1 -a TGGAATTCTCGG reads/raw/SRR6908979.fastq.gz\n",
      "Processing single-end reads on 40 cores ...\n",
      "Finished in 34.86 s (2 µs/read; 38.41 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              22,318,264\n",
      "Reads with adapters:                22,153,899 (99.3%)\n",
      "Reads written (passing filters):    22,318,264 (100.0%)\n",
      "\n",
      "Total basepairs processed: 1,138,231,464 bp\n",
      "Quality-trimmed:               1,885,439 bp (0.2%)\n",
      "Total written (filtered):    605,452,316 bp (53.2%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: TGGAATTCTCGG; Type: regular 3'; Length: 12; Trimmed: 22153899 times\n",
      "\n",
      "Minimum overlap: 1\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-12 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 13.8%\n",
      "  C: 53.7%\n",
      "  G: 10.9%\n",
      "  T: 21.4%\n",
      "  none/other: 0.3%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t51942\t5579566.0\t0\t51942\n",
      "2\t16724\t1394891.5\t0\t16724\n",
      "3\t12626\t348722.9\t0\t12626\n",
      "4\t11534\t87180.7\t0\t11534\n",
      "5\t17841\t21795.2\t0\t17841\n",
      "6\t21614\t5448.8\t0\t21614\n",
      "7\t36928\t1362.2\t0\t36928\n",
      "8\t81389\t340.5\t0\t81389\n",
      "9\t110577\t85.1\t0\t110107 470\n",
      "10\t193891\t21.3\t1\t187248 6643\n",
      "11\t284286\t5.3\t1\t278286 6000\n",
      "12\t210717\t1.3\t1\t205922 4795\n",
      "13\t223363\t1.3\t1\t216903 6460\n",
      "14\t248651\t1.3\t1\t242819 5832\n",
      "15\t364015\t1.3\t1\t355358 8657\n",
      "16\t351771\t1.3\t1\t342646 9125\n",
      "17\t871852\t1.3\t1\t851320 20532\n",
      "18\t567202\t1.3\t1\t551145 16057\n",
      "19\t676437\t1.3\t1\t655907 20530\n",
      "20\t1072265\t1.3\t1\t1045895 26370\n",
      "21\t2167867\t1.3\t1\t2131154 36713\n",
      "22\t2025289\t1.3\t1\t1988751 36538\n",
      "23\t1906958\t1.3\t1\t1868230 38728\n",
      "24\t1488439\t1.3\t1\t1460743 27696\n",
      "25\t1112257\t1.3\t1\t1092100 20157\n",
      "26\t1137815\t1.3\t1\t1114489 23326\n",
      "27\t1078157\t1.3\t1\t1053126 25031\n",
      "28\t719943\t1.3\t1\t704212 15731\n",
      "29\t880415\t1.3\t1\t865040 15375\n",
      "30\t706561\t1.3\t1\t694066 12495\n",
      "31\t498357\t1.3\t1\t488195 10162\n",
      "32\t905876\t1.3\t1\t890005 15871\n",
      "33\t400166\t1.3\t1\t391922 8244\n",
      "34\t387049\t1.3\t1\t378887 8162\n",
      "35\t260215\t1.3\t1\t254903 5312\n",
      "36\t209489\t1.3\t1\t204902 4587\n",
      "37\t152569\t1.3\t1\t149223 3346\n",
      "38\t148248\t1.3\t1\t145054 3194\n",
      "39\t220877\t1.3\t1\t216433 4444\n",
      "40\t112139\t1.3\t1\t109862 2277\n",
      "41\t63556\t1.3\t1\t62296 1260\n",
      "42\t42885\t1.3\t1\t42040 845\n",
      "43\t28316\t1.3\t1\t27790 526\n",
      "44\t11147\t1.3\t1\t10916 231\n",
      "45\t4004\t1.3\t1\t3924 80\n",
      "46\t1487\t1.3\t1\t1450 37\n",
      "47\t415\t1.3\t1\t403 12\n",
      "48\t542\t1.3\t1\t516 26\n",
      "49\t963\t1.3\t1\t922 41\n",
      "50\t3267\t1.3\t1\t3135 132\n",
      "51\t53006\t1.3\t1\t51868 1138\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: reads/raw/SRR6908979.fastq.gz\n",
      "=============================================\n",
      "22318264 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 18 bp:\t1733686 (7.8%)\n",
      "\n",
      "Using an excessive number of cores has a diminishing return! It is recommended not to exceed 8 cores per trimming process (you asked for 40 cores). Please consider re-specifying\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 4.1\n",
      "Could not detect version of Python used by Cutadapt from the first line of Cutadapt (but found this: >>>#!/bin/sh<<<)\n",
      "Letting the (modified) Cutadapt deal with the Python version instead\n",
      "pigz 2.6\n",
      "Parallel gzip (pigz) detected. Proceeding with multicore (de)compression using 40 cores\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Output will be written into the directory: /home/steve/Dropbox/bioinf/gih/reads/processed/\n",
      "Reducing length cutoff to 18bp for small RNA-Seq reads because a cutoff of 20bp may remove some short species of small RNAs if they had been trimmed by 1,2 or 3bp\n",
      "Writing report to '/home/steve/Dropbox/bioinf/gih/reads/processed/SRR6908989.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: reads/raw/SRR6908989.fastq.gz\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.6.7\n",
      "Cutadapt version: 4.1\n",
      "Python version: could not detect\n",
      "Number of cores used for trimming: 40\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'TGGAATTCTCGG' (Illumina small RNA adapter; user defined)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 18 bp\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Cutadapt seems to be fairly up-to-date (version 4.1). Setting -j 40\n",
      "Writing final adapter and quality trimmed output to SRR6908989_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'TGGAATTCTCGG' from file reads/raw/SRR6908989.fastq.gz <<< \n",
      "This is cutadapt 4.1 with Python 3.9.13\n",
      "Command line parameters: -j 40 -e 0.1 -q 20 -O 1 -a TGGAATTCTCGG reads/raw/SRR6908989.fastq.gz\n",
      "Processing single-end reads on 40 cores ...\n",
      "Finished in 12.62 s (2 µs/read; 35.16 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:               7,396,305\n",
      "Reads with adapters:                 7,293,575 (98.6%)\n",
      "Reads written (passing filters):     7,396,305 (100.0%)\n",
      "\n",
      "Total basepairs processed:   377,211,555 bp\n",
      "Quality-trimmed:                 488,279 bp (0.1%)\n",
      "Total written (filtered):    226,144,854 bp (60.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: TGGAATTCTCGG; Type: regular 3'; Length: 12; Trimmed: 7293575 times\n",
      "\n",
      "Minimum overlap: 1\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-12 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 20.5%\n",
      "  C: 41.2%\n",
      "  G: 13.7%\n",
      "  T: 24.2%\n",
      "  none/other: 0.4%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t71391\t1849076.2\t0\t71391\n",
      "2\t47054\t462269.1\t0\t47054\n",
      "3\t53357\t115567.3\t0\t53357\n",
      "4\t56115\t28891.8\t0\t56115\n",
      "5\t69307\t7223.0\t0\t69307\n",
      "6\t74609\t1805.7\t0\t74609\n",
      "7\t85300\t451.4\t0\t85300\n",
      "8\t105881\t112.9\t0\t105881\n",
      "9\t134551\t28.2\t0\t134088 463\n",
      "10\t241004\t7.1\t1\t237119 3885\n",
      "11\t215671\t1.8\t1\t212482 3189\n",
      "12\t156639\t0.4\t1\t153786 2853\n",
      "13\t184568\t0.4\t1\t181232 3336\n",
      "14\t167606\t0.4\t1\t164964 2642\n",
      "15\t228835\t0.4\t1\t225083 3752\n",
      "16\t230546\t0.4\t1\t225672 4874\n",
      "17\t892364\t0.4\t1\t879972 12392\n",
      "18\t277487\t0.4\t1\t272847 4640\n",
      "19\t252396\t0.4\t1\t247902 4494\n",
      "20\t241423\t0.4\t1\t237351 4072\n",
      "21\t305021\t0.4\t1\t300690 4331\n",
      "22\t301929\t0.4\t1\t297452 4477\n",
      "23\t315608\t0.4\t1\t310501 5107\n",
      "24\t271327\t0.4\t1\t267002 4325\n",
      "25\t235274\t0.4\t1\t231701 3573\n",
      "26\t211722\t0.4\t1\t208163 3559\n",
      "27\t222441\t0.4\t1\t218226 4215\n",
      "28\t204397\t0.4\t1\t200829 3568\n",
      "29\t223431\t0.4\t1\t220082 3349\n",
      "30\t192021\t0.4\t1\t189042 2979\n",
      "31\t149327\t0.4\t1\t146901 2426\n",
      "32\t177590\t0.4\t1\t174761 2829\n",
      "33\t130261\t0.4\t1\t128006 2255\n",
      "34\t109218\t0.4\t1\t107540 1678\n",
      "35\t75347\t0.4\t1\t74011 1336\n",
      "36\t73665\t0.4\t1\t72313 1352\n",
      "37\t57901\t0.4\t1\t56871 1030\n",
      "38\t51814\t0.4\t1\t50915 899\n",
      "39\t76015\t0.4\t1\t74849 1166\n",
      "40\t33510\t0.4\t1\t32961 549\n",
      "41\t18740\t0.4\t1\t18438 302\n",
      "42\t16415\t0.4\t1\t16169 246\n",
      "43\t13282\t0.4\t1\t13058 224\n",
      "44\t6545\t0.4\t1\t6420 125\n",
      "45\t2445\t0.4\t1\t2402 43\n",
      "46\t1119\t0.4\t1\t1096 23\n",
      "47\t586\t0.4\t1\t567 19\n",
      "48\t500\t0.4\t1\t484 16\n",
      "49\t1017\t0.4\t1\t983 34\n",
      "50\t1829\t0.4\t1\t1779 50\n",
      "51\t27174\t0.4\t1\t26680 494\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "RUN STATISTICS FOR INPUT FILE: reads/raw/SRR6908989.fastq.gz\n",
      "=============================================\n",
      "7396305 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 18 bp:\t579179 (7.8%)\n",
      "\n",
      "Using an excessive number of cores has a diminishing return! It is recommended not to exceed 8 cores per trimming process (you asked for 40 cores). Please consider re-specifying\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 4.1\n",
      "Could not detect version of Python used by Cutadapt from the first line of Cutadapt (but found this: >>>#!/bin/sh<<<)\n",
      "Letting the (modified) Cutadapt deal with the Python version instead\n",
      "pigz 2.6\n",
      "Parallel gzip (pigz) detected. Proceeding with multicore (de)compression using 40 cores\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Output will be written into the directory: /home/steve/Dropbox/bioinf/gih/reads/processed/\n",
      "Reducing length cutoff to 18bp for small RNA-Seq reads because a cutoff of 20bp may remove some short species of small RNAs if they had been trimmed by 1,2 or 3bp\n",
      "Writing report to '/home/steve/Dropbox/bioinf/gih/reads/processed/SRR6908978.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: reads/raw/SRR6908978.fastq.gz\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.6.7\n",
      "Cutadapt version: 4.1\n",
      "Python version: could not detect\n",
      "Number of cores used for trimming: 40\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'TGGAATTCTCGG' (Illumina small RNA adapter; user defined)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 18 bp\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Cutadapt seems to be fairly up-to-date (version 4.1). Setting -j 40\n",
      "Writing final adapter and quality trimmed output to SRR6908978_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'TGGAATTCTCGG' from file reads/raw/SRR6908978.fastq.gz <<< \n",
      "10000000 sequences processed\n",
      "20000000 sequences processed\n",
      "This is cutadapt 4.1 with Python 3.9.13\n",
      "Command line parameters: -j 40 -e 0.1 -q 20 -O 1 -a TGGAATTCTCGG reads/raw/SRR6908978.fastq.gz\n",
      "Processing single-end reads on 40 cores ...\n",
      "Finished in 34.68 s (2 µs/read; 37.87 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              21,884,838\n",
      "Reads with adapters:                21,702,124 (99.2%)\n",
      "Reads written (passing filters):    21,884,838 (100.0%)\n",
      "\n",
      "Total basepairs processed: 1,116,126,738 bp\n",
      "Quality-trimmed:               1,305,023 bp (0.1%)\n",
      "Total written (filtered):    598,272,382 bp (53.6%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: TGGAATTCTCGG; Type: regular 3'; Length: 12; Trimmed: 21702124 times\n",
      "\n",
      "Minimum overlap: 1\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-12 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 16.2%\n",
      "  C: 46.1%\n",
      "  G: 13.2%\n",
      "  T: 24.4%\n",
      "  none/other: 0.1%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t96348\t5471209.5\t0\t96348\n",
      "2\t49383\t1367802.4\t0\t49383\n",
      "3\t56665\t341950.6\t0\t56665\n",
      "4\t58668\t85487.6\t0\t58668\n",
      "5\t73724\t21371.9\t0\t73724\n",
      "6\t93256\t5343.0\t0\t93256\n",
      "7\t127369\t1335.7\t0\t127369\n",
      "8\t213260\t333.9\t0\t213260\n",
      "9\t234751\t83.5\t0\t233824 927\n",
      "10\t315487\t20.9\t1\t309623 5864\n",
      "11\t456070\t5.2\t1\t448307 7763\n",
      "12\t314307\t1.3\t1\t308221 6086\n",
      "13\t313249\t1.3\t1\t306493 6756\n",
      "14\t308682\t1.3\t1\t302965 5717\n",
      "15\t422586\t1.3\t1\t414466 8120\n",
      "16\t397756\t1.3\t1\t390066 7690\n",
      "17\t892416\t1.3\t1\t874559 17857\n",
      "18\t560784\t1.3\t1\t548555 12229\n",
      "19\t672227\t1.3\t1\t656531 15696\n",
      "20\t963216\t1.3\t1\t943215 20001\n",
      "21\t1568694\t1.3\t1\t1540381 28313\n",
      "22\t1420019\t1.3\t1\t1393150 26869\n",
      "23\t1349485\t1.3\t1\t1321622 27863\n",
      "24\t1131890\t1.3\t1\t1110204 21686\n",
      "25\t945593\t1.3\t1\t927494 18099\n",
      "26\t902170\t1.3\t1\t881940 20230\n",
      "27\t951540\t1.3\t1\t928534 23006\n",
      "28\t697930\t1.3\t1\t682170 15760\n",
      "29\t834412\t1.3\t1\t817452 16960\n",
      "30\t747746\t1.3\t1\t732670 15076\n",
      "31\t566481\t1.3\t1\t555006 11475\n",
      "32\t821294\t1.3\t1\t806393 14901\n",
      "33\t552036\t1.3\t1\t541177 10859\n",
      "34\t511824\t1.3\t1\t502101 9723\n",
      "35\t368245\t1.3\t1\t361067 7178\n",
      "36\t327132\t1.3\t1\t320422 6710\n",
      "37\t262507\t1.3\t1\t257166 5341\n",
      "38\t260384\t1.3\t1\t255320 5064\n",
      "39\t392574\t1.3\t1\t385899 6675\n",
      "40\t201597\t1.3\t1\t197927 3670\n",
      "41\t105488\t1.3\t1\t103595 1893\n",
      "42\t66834\t1.3\t1\t65661 1173\n",
      "43\t40871\t1.3\t1\t40102 769\n",
      "44\t18548\t1.3\t1\t18183 365\n",
      "45\t6647\t1.3\t1\t6495 152\n",
      "46\t2355\t1.3\t1\t2304 51\n",
      "47\t620\t1.3\t1\t610 10\n",
      "48\t521\t1.3\t1\t506 15\n",
      "49\t1075\t1.3\t1\t1041 34\n",
      "50\t1682\t1.3\t1\t1593 89\n",
      "51\t23726\t1.3\t1\t23042 684\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: reads/raw/SRR6908978.fastq.gz\n",
      "=============================================\n",
      "21884838 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 18 bp:\t2625759 (12.0%)\n",
      "\n",
      "Using an excessive number of cores has a diminishing return! It is recommended not to exceed 8 cores per trimming process (you asked for 40 cores). Please consider re-specifying\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt version: 4.1\n",
      "Could not detect version of Python used by Cutadapt from the first line of Cutadapt (but found this: >>>#!/bin/sh<<<)\n",
      "Letting the (modified) Cutadapt deal with the Python version instead\n",
      "pigz 2.6\n",
      "Parallel gzip (pigz) detected. Proceeding with multicore (de)compression using 40 cores\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Output will be written into the directory: /home/steve/Dropbox/bioinf/gih/reads/processed/\n",
      "Reducing length cutoff to 18bp for small RNA-Seq reads because a cutoff of 20bp may remove some short species of small RNAs if they had been trimmed by 1,2 or 3bp\n",
      "Writing report to '/home/steve/Dropbox/bioinf/gih/reads/processed/SRR6908984.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: reads/raw/SRR6908984.fastq.gz\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.6.7\n",
      "Cutadapt version: 4.1\n",
      "Python version: could not detect\n",
      "Number of cores used for trimming: 40\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'TGGAATTCTCGG' (Illumina small RNA adapter; user defined)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 18 bp\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Cutadapt seems to be fairly up-to-date (version 4.1). Setting -j 40\n",
      "Writing final adapter and quality trimmed output to SRR6908984_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'TGGAATTCTCGG' from file reads/raw/SRR6908984.fastq.gz <<< \n",
      "This is cutadapt 4.1 with Python 3.9.13\n",
      "Command line parameters: -j 40 -e 0.1 -q 20 -O 1 -a TGGAATTCTCGG reads/raw/SRR6908984.fastq.gz\n",
      "Processing single-end reads on 40 cores ...\n",
      "Finished in 11.91 s (2 µs/read; 34.65 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:               6,877,929\n",
      "Reads with adapters:                 6,771,149 (98.4%)\n",
      "Reads written (passing filters):     6,877,929 (100.0%)\n",
      "\n",
      "Total basepairs processed:   350,774,379 bp\n",
      "Quality-trimmed:                 483,813 bp (0.1%)\n",
      "Total written (filtered):    208,003,932 bp (59.3%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: TGGAATTCTCGG; Type: regular 3'; Length: 12; Trimmed: 6771149 times\n",
      "\n",
      "Minimum overlap: 1\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-12 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 19.9%\n",
      "  C: 39.8%\n",
      "  G: 14.9%\n",
      "  T: 24.7%\n",
      "  none/other: 0.6%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t80266\t1719482.2\t0\t80266\n",
      "2\t53970\t429870.6\t0\t53970\n",
      "3\t59194\t107467.6\t0\t59194\n",
      "4\t60274\t26866.9\t0\t60274\n",
      "5\t71701\t6716.7\t0\t71701\n",
      "6\t76212\t1679.2\t0\t76212\n",
      "7\t87652\t419.8\t0\t87652\n",
      "8\t107883\t104.9\t0\t107883\n",
      "9\t135360\t26.2\t0\t134752 608\n",
      "10\t210498\t6.6\t1\t206893 3605\n",
      "11\t203894\t1.6\t1\t200489 3405\n",
      "12\t155772\t0.4\t1\t152502 3270\n",
      "13\t170099\t0.4\t1\t166853 3246\n",
      "14\t154847\t0.4\t1\t152279 2568\n",
      "15\t203304\t0.4\t1\t199933 3371\n",
      "16\t187223\t0.4\t1\t183437 3786\n",
      "17\t470633\t0.4\t1\t462772 7861\n",
      "18\t220989\t0.4\t1\t216680 4309\n",
      "19\t215825\t0.4\t1\t211560 4265\n",
      "20\t231776\t0.4\t1\t227263 4513\n",
      "21\t347732\t0.4\t1\t342112 5620\n",
      "22\t330486\t0.4\t1\t325096 5390\n",
      "23\t326489\t0.4\t1\t320849 5640\n",
      "24\t277413\t0.4\t1\t272496 4917\n",
      "25\t231676\t0.4\t1\t227798 3878\n",
      "26\t218046\t0.4\t1\t214131 3915\n",
      "27\t235012\t0.4\t1\t230265 4747\n",
      "28\t197650\t0.4\t1\t193678 3972\n",
      "29\t260348\t0.4\t1\t255921 4427\n",
      "30\t194103\t0.4\t1\t190794 3309\n",
      "31\t144305\t0.4\t1\t141688 2617\n",
      "32\t177296\t0.4\t1\t174298 2998\n",
      "33\t110634\t0.4\t1\t108647 1987\n",
      "34\t104036\t0.4\t1\t102151 1885\n",
      "35\t70211\t0.4\t1\t68738 1473\n",
      "36\t64372\t0.4\t1\t63074 1298\n",
      "37\t54953\t0.4\t1\t53960 993\n",
      "38\t42715\t0.4\t1\t41865 850\n",
      "39\t86579\t0.4\t1\t85081 1498\n",
      "40\t35446\t0.4\t1\t34817 629\n",
      "41\t19368\t0.4\t1\t19015 353\n",
      "42\t16833\t0.4\t1\t16551 282\n",
      "43\t13253\t0.4\t1\t13039 214\n",
      "44\t5926\t0.4\t1\t5830 96\n",
      "45\t1936\t0.4\t1\t1892 44\n",
      "46\t1058\t0.4\t1\t1039 19\n",
      "47\t693\t0.4\t1\t678 15\n",
      "48\t662\t0.4\t1\t640 22\n",
      "49\t1466\t0.4\t1\t1419 47\n",
      "50\t2477\t0.4\t1\t2381 96\n",
      "51\t40603\t0.4\t1\t39726 877\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: reads/raw/SRR6908984.fastq.gz\n",
      "=============================================\n",
      "6877929 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 18 bp:\t567088 (8.2%)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "import glob\n",
    "for i in glob.glob(\"reads/raw/*.gz\"):\n",
    "    !trim_galore --small_rna -o reads/processed -j 40 {i}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b0dc7d90",
   "metadata": {},
   "source": [
    "## Align"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "55b106e3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "reads/processed/SRR6908980_trimmed.fq.gz,reads/processed/SRR6908989_trimmed.fq.gz,reads/processed/SRR6908982_trimmed.fq.gz,reads/processed/SRR6908987_trimmed.fq.gz,reads/processed/SRR6908983_trimmed.fq.gz,reads/processed/SRR6908981_trimmed.fq.gz,reads/processed/SRR6908988_trimmed.fq.gz,reads/processed/SRR6908979_trimmed.fq.gz,reads/processed/SRR6908978_trimmed.fq.gz,reads/processed/SRR6908985_trimmed.fq.gz,reads/processed/SRR6908984_trimmed.fq.gz,reads/processed/SRR6908986_trimmed.fq.gz\n"
     ]
    }
   ],
   "source": [
    "reads = \"\"\n",
    "for i in glob.glob('reads/processed/*.gz'):\n",
    "    reads = reads + i + \",\"\n",
    "reads = reads[:-1]\n",
    "print(reads)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "c3c695e4",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Loading individual read counts\n",
      "\n",
      "SCRAM is attempting to load read files in FASTQ format\n",
      "reads/processed/SRR6908984_trimmed.fq.gz - 3,511,950 reads processed\n",
      "reads/processed/SRR6908989_trimmed.fq.gz - 3,438,998 reads processed\n",
      "reads/processed/SRR6908983_trimmed.fq.gz - 4,808,626 reads processed\n",
      "reads/processed/SRR6908982_trimmed.fq.gz - 4,390,544 reads processed\n",
      "reads/processed/SRR6908980_trimmed.fq.gz - 5,905,274 reads processed\n",
      "reads/processed/SRR6908987_trimmed.fq.gz - 4,270,584 reads processed\n",
      "reads/processed/SRR6908988_trimmed.fq.gz - 6,439,766 reads processed\n",
      "reads/processed/SRR6908986_trimmed.fq.gz - 8,627,707 reads processed\n",
      "reads/processed/SRR6908981_trimmed.fq.gz - 12,583,592 reads processed\n",
      "reads/processed/SRR6908985_trimmed.fq.gz - 11,014,955 reads processed\n",
      "reads/processed/SRR6908978_trimmed.fq.gz - 14,142,603 reads processed\n",
      "reads/processed/SRR6908979_trimmed.fq.gz - 16,830,158 reads processed\n",
      "\n",
      "Loading reference\n",
      "No. of reference sequences:  1\n",
      "Combined length of reference sequences: 1,247 nt\n",
      "\n",
      "Aligning 20 nt reads\n",
      "\n",
      "Aligning 21 nt reads\n",
      "\n",
      "Aligning 22 nt reads\n",
      "\n",
      "Aligning 23 nt reads\n",
      "\n",
      "Aligning 24 nt reads\n",
      "\n",
      "Alignment complete.  Total time taken = 18.878004956s\n"
     ]
    }
   ],
   "source": [
    "! software/scram2 profile \\\n",
    "    -r ref/tri6.fa \\\n",
    "    -1 {reads} \\\n",
    "    -l 20,21,22,23,24 \\\n",
    "    -o test_align"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "920c1c08",
   "metadata": {},
   "source": [
    "## Check the 21nt alignments"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "8cdb13bd",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Header</th>\n",
       "      <th>len</th>\n",
       "      <th>sRNA</th>\n",
       "      <th>Position</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Times aligned</th>\n",
       "      <th>SRR6908984_trimmed.fq.gz</th>\n",
       "      <th>SRR6908989_trimmed.fq.gz</th>\n",
       "      <th>SRR6908983_trimmed.fq.gz</th>\n",
       "      <th>SRR6908982_trimmed.fq.gz</th>\n",
       "      <th>SRR6908980_trimmed.fq.gz</th>\n",
       "      <th>SRR6908987_trimmed.fq.gz</th>\n",
       "      <th>SRR6908988_trimmed.fq.gz</th>\n",
       "      <th>SRR6908986_trimmed.fq.gz</th>\n",
       "      <th>SRR6908981_trimmed.fq.gz</th>\n",
       "      <th>SRR6908985_trimmed.fq.gz</th>\n",
       "      <th>SRR6908978_trimmed.fq.gz</th>\n",
       "      <th>SRR6908979_trimmed.fq.gz</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AB017495.1 Fusarium graminearum Tri6 gene for ...</td>\n",
       "      <td>1247</td>\n",
       "      <td>TTCGTCCTCCATGTAAATCAT</td>\n",
       "      <td>396</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.091</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AB017495.1 Fusarium graminearum Tri6 gene for ...</td>\n",
       "      <td>1247</td>\n",
       "      <td>TCCAAGATTCGTAGTGAGATT</td>\n",
       "      <td>415</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.091</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AB017495.1 Fusarium graminearum Tri6 gene for ...</td>\n",
       "      <td>1247</td>\n",
       "      <td>TCTCACTACGAATCTTGGAGC</td>\n",
       "      <td>417</td>\n",
       "      <td>+</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.234</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.182</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AB017495.1 Fusarium graminearum Tri6 gene for ...</td>\n",
       "      <td>1247</td>\n",
       "      <td>CGCTCCAAGATTCGTAGTGAG</td>\n",
       "      <td>418</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.091</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AB017495.1 Fusarium graminearum Tri6 gene for ...</td>\n",
       "      <td>1247</td>\n",
       "      <td>TCACTACGAATCTTGGAGCGC</td>\n",
       "      <td>419</td>\n",
       "      <td>+</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.182</td>\n",
       "      <td>0.283</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334</th>\n",
       "      <td>AB017495.1 Fusarium graminearum Tri6 gene for ...</td>\n",
       "      <td>1247</td>\n",
       "      <td>TCCGCCTATAGTGATCTCGCA</td>\n",
       "      <td>1018</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.119</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>335</th>\n",
       "      <td>AB017495.1 Fusarium graminearum Tri6 gene for ...</td>\n",
       "      <td>1247</td>\n",
       "      <td>TGCGAGATCACTATAGGCGGA</td>\n",
       "      <td>1018</td>\n",
       "      <td>+</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.397</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>7.724</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>336</th>\n",
       "      <td>AB017495.1 Fusarium graminearum Tri6 gene for ...</td>\n",
       "      <td>1247</td>\n",
       "      <td>GCGAGATCACTATAGGCGGAT</td>\n",
       "      <td>1019</td>\n",
       "      <td>+</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.119</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>337</th>\n",
       "      <td>AB017495.1 Fusarium graminearum Tri6 gene for ...</td>\n",
       "      <td>1247</td>\n",
       "      <td>TATCCGCCTATAGTGATCTCG</td>\n",
       "      <td>1020</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.059</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>338</th>\n",
       "      <td>AB017495.1 Fusarium graminearum Tri6 gene for ...</td>\n",
       "      <td>1247</td>\n",
       "      <td>CGAGATCACTATAGGCGGATA</td>\n",
       "      <td>1020</td>\n",
       "      <td>+</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.116</td>\n",
       "      <td>1.987</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>25.906</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>339 rows × 18 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                Header   len  \\\n",
       "0    AB017495.1 Fusarium graminearum Tri6 gene for ...  1247   \n",
       "1    AB017495.1 Fusarium graminearum Tri6 gene for ...  1247   \n",
       "2    AB017495.1 Fusarium graminearum Tri6 gene for ...  1247   \n",
       "3    AB017495.1 Fusarium graminearum Tri6 gene for ...  1247   \n",
       "4    AB017495.1 Fusarium graminearum Tri6 gene for ...  1247   \n",
       "..                                                 ...   ...   \n",
       "334  AB017495.1 Fusarium graminearum Tri6 gene for ...  1247   \n",
       "335  AB017495.1 Fusarium graminearum Tri6 gene for ...  1247   \n",
       "336  AB017495.1 Fusarium graminearum Tri6 gene for ...  1247   \n",
       "337  AB017495.1 Fusarium graminearum Tri6 gene for ...  1247   \n",
       "338  AB017495.1 Fusarium graminearum Tri6 gene for ...  1247   \n",
       "\n",
       "                      sRNA  Position Strand  Times aligned  \\\n",
       "0    TTCGTCCTCCATGTAAATCAT       396      -              1   \n",
       "1    TCCAAGATTCGTAGTGAGATT       415      -              1   \n",
       "2    TCTCACTACGAATCTTGGAGC       417      +              1   \n",
       "3    CGCTCCAAGATTCGTAGTGAG       418      -              1   \n",
       "4    TCACTACGAATCTTGGAGCGC       419      +              1   \n",
       "..                     ...       ...    ...            ...   \n",
       "334  TCCGCCTATAGTGATCTCGCA      1018      -              1   \n",
       "335  TGCGAGATCACTATAGGCGGA      1018      +              1   \n",
       "336  GCGAGATCACTATAGGCGGAT      1019      +              1   \n",
       "337  TATCCGCCTATAGTGATCTCG      1020      -              1   \n",
       "338  CGAGATCACTATAGGCGGATA      1020      +              1   \n",
       "\n",
       "     SRR6908984_trimmed.fq.gz  SRR6908989_trimmed.fq.gz  \\\n",
       "0                         0.0                       0.0   \n",
       "1                         0.0                       0.0   \n",
       "2                         0.0                       0.0   \n",
       "3                         0.0                       0.0   \n",
       "4                         0.0                       0.0   \n",
       "..                        ...                       ...   \n",
       "334                       0.0                       0.0   \n",
       "335                       0.0                       0.0   \n",
       "336                       0.0                       0.0   \n",
       "337                       0.0                       0.0   \n",
       "338                       0.0                       0.0   \n",
       "\n",
       "     SRR6908983_trimmed.fq.gz  SRR6908982_trimmed.fq.gz  \\\n",
       "0                         0.0                       0.0   \n",
       "1                         0.0                       0.0   \n",
       "2                         0.0                       0.0   \n",
       "3                         0.0                       0.0   \n",
       "4                         0.0                       0.0   \n",
       "..                        ...                       ...   \n",
       "334                       0.0                       0.0   \n",
       "335                       0.0                       0.0   \n",
       "336                       0.0                       0.0   \n",
       "337                       0.0                       0.0   \n",
       "338                       0.0                       0.0   \n",
       "\n",
       "     SRR6908980_trimmed.fq.gz  SRR6908987_trimmed.fq.gz  \\\n",
       "0                         0.0                     0.000   \n",
       "1                         0.0                     0.000   \n",
       "2                         0.0                     0.234   \n",
       "3                         0.0                     0.000   \n",
       "4                         0.0                     0.000   \n",
       "..                        ...                       ...   \n",
       "334                       0.0                     0.000   \n",
       "335                       0.0                     0.000   \n",
       "336                       0.0                     0.000   \n",
       "337                       0.0                     0.000   \n",
       "338                       0.0                     0.000   \n",
       "\n",
       "     SRR6908988_trimmed.fq.gz  SRR6908986_trimmed.fq.gz  \\\n",
       "0                         0.0                     0.000   \n",
       "1                         0.0                     0.000   \n",
       "2                         0.0                     0.000   \n",
       "3                         0.0                     0.000   \n",
       "4                         0.0                     0.000   \n",
       "..                        ...                       ...   \n",
       "334                       0.0                     0.000   \n",
       "335                       0.0                     0.000   \n",
       "336                       0.0                     0.000   \n",
       "337                       0.0                     0.000   \n",
       "338                       0.0                     0.116   \n",
       "\n",
       "     SRR6908981_trimmed.fq.gz  SRR6908985_trimmed.fq.gz  \\\n",
       "0                       0.000                     0.091   \n",
       "1                       0.000                     0.091   \n",
       "2                       0.000                     0.182   \n",
       "3                       0.000                     0.091   \n",
       "4                       0.000                     0.182   \n",
       "..                        ...                       ...   \n",
       "334                     0.000                     0.000   \n",
       "335                     0.397                     0.000   \n",
       "336                     0.000                     0.000   \n",
       "337                     0.000                     0.000   \n",
       "338                     1.987                     0.000   \n",
       "\n",
       "     SRR6908978_trimmed.fq.gz  SRR6908979_trimmed.fq.gz  \n",
       "0                       0.000                     0.000  \n",
       "1                       0.000                     0.000  \n",
       "2                       0.000                     0.000  \n",
       "3                       0.000                     0.000  \n",
       "4                       0.283                     0.000  \n",
       "..                        ...                       ...  \n",
       "334                     0.000                     0.119  \n",
       "335                     0.000                     7.724  \n",
       "336                     0.000                     0.119  \n",
       "337                     0.000                     0.059  \n",
       "338                     0.000                    25.906  \n",
       "\n",
       "[339 rows x 18 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "align_21 = pd.read_csv(\"test_align_21.csv\")\n",
    "align_21"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bf563a7a",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
